### PR TITLE
Add two vignettes: tutorial and performance

### LINKED ISSUE #4: Performance comparison
In order to show the interest of having another package to compute functional diversity indices, we should also make a vignette that shows performance comparison between `fundiversity` and related packages.

### PR DESCRIPTION
**Pull request by Rekyt**
To fix #4 and write a tutorial on the indices as well as the use of the package.

### CODE REVIEW

**Comment by Bisaloo** [comment]
This might belong to a later PR but it might be nice to add plots for `microbenchmark()` runs

File: vignettes/fundiversity.Rmd [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+```{r raoq-dissim}
+# Compute dissimilarity between species with the Manhattan distance
+trait_dissim <- dist(traits_birds, method = "manhattan")
+
+fd_raoq(traits = NULL, dist_matrix = trait_dissim)
+
+fd_raoq(traits = NULL, site_sp_birds, dist_matrix = as.matrix(trait_dissim))
+```
+**NB**: If you do provide a dissimilarity matrix through the `dist_matrix` argument, set up the `traits` argument to `NULL` explicitly with `traits = NULL`.
```
Comment (Bisaloo) [inline vignettes/fundiversity.Rmd:184]: This seems complex and suboptimal. Should we just add `traits = NULL` as default?

Comment (Rekyt) [reply vignettes/fundiversity.Rmd:184]: Indeed, we could add traits = NULL but should also add a check about it.

File: vignettes/performance.Rmd [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+  species_200    = fd_raoq(     traits_birds, site_sp_birds),
+  species_1000   = fd_raoq(  null_sp_1000$tr, null_sp_1000$si),
+  species_10000  = fd_raoq( null_sp_10000$tr, null_sp_10000$si),
+  times = 100
+)
+#> Unit: milliseconds
+#>           expr         min           lq         mean       median           uq        max neval
+#>    species_200    2.325501     2.698201     8.135771     2.904751     3.276052   258.5704   100
+#>   species_1000   62.716601    75.065652    84.869521    81.662751    87.427851   267.6919   100
+#>  species_10000 9884.199601 12087.106450 12954.704923 12702.614551 13878.859601 17256.5673   100
```
Comment (Bisaloo) [inline vignettes/performance.Rmd:341]: The increase in computation time with the number of species here is very strange. It may be worth investigating.

Comment (Rekyt) [reply vignettes/performance.Rmd:341]: I think it's probably due to the overhead of manipulating a bigger distance matrix. But it's true that it's striking between 1000 and 10000 species.

File: vignettes/fundiversity.Rmd (hunk 1 above)
Comment (Bisaloo) [reply vignettes/fundiversity.Rmd:184]: Correct me if I'm wrong but the current check seems sufficient in my opinion

**Pull request merged**